### PR TITLE
Update: NuGet packages, Add: ConfigReader & Default config data

### PR DESCRIPTION
**Pull request by gallayl**

### CODE REVIEW

File: RepoSync/MemoryProvider/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -1,5 +1,4 @@
 ﻿using System.Reflection;
-using System.Runtime.CompilerServices;
```
Comment (marosvolgyiz) [inline RepoSync/MemoryProvider/Properties/AssemblyInfo.cs:2]: What is that? Why not use the CompilerServices?

File: RepoSync/FileSystemProvider/FileSystemProvider.cs [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
+using SenseNet.Client;
 
 namespace FileSystemProvider
 {
     public class FileSystemProvider : IRepoSyncProvider
     {
+        public List<string> RequiredOptions => new List<string> { "Path" };
+        public Dictionary<string, string> Settings { get; set; }
+        public IRepoSyncFilter Filter { get; set; }
+        public Task<Content> LoadAsync(string path)
```
Comment (marosvolgyiz) [inline RepoSync/FileSystemProvider/FileSystemProvider.cs]: Use the async keyword for the asyncronous methods.

File: RepoSync/RepoSync/IRepoSyncProvider.cs [hunk 3]
```diff
@@ -9,11 +9,13 @@ namespace RepoSync
     /// </summary>
     public interface IRepoSyncProvider
     {
+
+        List<string> RequiredOptions { get; }
```
Comment (marosvolgyiz) [inline RepoSync/RepoSync/IRepoSyncProvider.cs]: Please change the Options part of this property to Settings.

File: RepoSync/RepoSyncCLI/App.config [hunk 4]
```diff
@@ -1,5 +1,15 @@
-﻿<?xml version="1.0" encoding="utf-8" ?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
+  <appSettings>
```
Comment (marosvolgyiz) [inline RepoSync/RepoSyncCLI/App.config]: I think this is a little better solution for store configurations: 
``` XML 
<appSettings>
	<!--RepoSync configuration-->
        <add key="SourceProviderName" value=""/>
	<!-- Options format: key=value;key2=value2;key3=value3 ...-->
       <add key="SourceProviderOptions" value=""/>
       <add key="TargetProviderName" value=""/>
        <add key="TargetProviderOptions" value=""/>
	<!--List of ActionTypes: Compare, Sync -->
        <add key="ActionType" value="Compare"/>
	<!--RepoSync configuration-->
  </appSettings>```

File: RepoSync/RepoSyncCLI/ConfigReader.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Diagnostics;
+using System.Linq;
+
+namespace RepoSync
```
Comment (marosvolgyiz) [inline RepoSync/RepoSyncCLI/ConfigReader.cs]: Change all the namespace in the RepoSyncCLI project RepoSync -> RepoSync.CLI

**Review by marosvolgyiz — CHANGES_REQUESTED** [review]
Please modify that commented stuff.

File: RepoSync/MemoryProvider/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (gallayl) [reply RepoSync/MemoryProvider/Properties/AssemblyInfo.cs:2]: It was an unused reference.

**Pull request merged**